File: ruby/bob/bob.rb
class Statement
  def initialize content
    @content = content.to_s
  end

  def silence?
    @content.empty?
  end

  def question?
    @content.end_with? '?'
  end

  def yelling?
    @content == @content.upcase
  end
end

class Bob
  def hey input
    answer_to Statement.new(input)
  end

  def answer_to statement
    if statement.silence?
      "Fine. Be that way!"
    elsif statement.yelling?
      "Woah, chill out!"
    elsif statement.question?
      'Sure.'
    else
      "Whatever."
    end
  end
end

File: javascript/bob/bob.js
function silent(message){
  return message === '';
}

function shouted(message){
  return message.toUpperCase() === message;
}

function questioning(message){
  return message.substr(-1) == '?';
}

Bob = function(){};
Bob.prototype.hey = function(message){
  if (silent(message))
    return 'Fine, be that way!';
  else if (shouted(message))
    return 'Woah, chill out!';
  else if (questioning(message))
    return "Sure.";
  else
    return "Whatever.";
};

module.exports = Bob;
